tcases-lib/src/main/java/org/cornutum/tcases/VarBinding.java

```diff
@@ -41,22 +41,42 @@ public VarBinding( String varName, String valueName)
    */
   public VarBinding( String varName, String varType, String valueName)
     {
-    setVar( varName);
-    setType( varType);
-    setValue( valueName);
-    setValueValid( true);
+    this( varName, varType, valueName, false);
     }
   
   /**
    * Creates a new VarBinding object.
    */
   public VarBinding( IVarDef varDef, VarValueDef valueDef)
     {
-    this( varDef.getPathName(), varDef.getType(), valueDef.getName());
+    this( varDef.getPathName(), varDef.getType(), valueDef.getName(), valueDef.isNA());
     setValueValid( valueDef.getType().isValid());
     setVarDef( varDef);
     }
   
+  /**
+   * Creates a new "not applicable" VarBinding object.
+   */
+  public static VarBinding notApplicable( String varName, String varType)
+    {
+    return new VarBinding( varName, varType, null, true);
+    }
+  
+  /**
+   * Creates a new VarBinding object.
+   */
+  private VarBinding( String varName, String varType, String valueName, boolean valueNA)
+    {
+    setVar( varName);
+    setType( varType);
+    if( !valueNA)
+      {
+      setValue( valueName);
```

Comment:
so the valueName method argument would be silently dropped based on the boolean. Even as a private method, this should maybe throw an exception if valueName is not null.